src/metabase/actions/execution.clj

```diff
@@ -60,9 +60,17 @@
     {:body {:message (or (ex-message ex) (tru "Error executing action."))}
      :status 500}))
 
+(defn- implicit-action-table
+  [card_id]
+  (let [card (db/select-one Card :id card_id)
+        {:keys [table-id]} (query/query->database-and-table-ids (:dataset_query card))]
+    (hydrate (db/select-one Table :id table-id) :fields)))
+
 (defn- execute-custom-action [action request-parameters]
-  (let [action-type (:type action)
-        destination-parameters-by-id (m/index-by :id (:parameters action))]
+  (let [{action-type :type}          action
+        destination-parameters-by-id (m/index-by :id (:parameters action))
+        db-id                        (:db_id (implicit-action-table (:model_id action)))
```

Comment:
This does a lot of work just to get the DB of a model. Can you grab the database ID from the card?